modules/Bio/EnsEMBL/Hive/DBSQL/AccumulatorAdaptor.pm

```diff
@@ -47,6 +47,32 @@ sub default_table_name {
 }
 
 
+sub slicer {    # take a slice of the hashref (if only we could inline in Perl!)
+    my ($self, $hashref, $fields) = @_;
+
+    my $overflow_limit = $self->overflow_limit();
+
+    return [ map { eval { my $value = $hashref->{$_};
+                          my $ol = $overflow_limit->{$_};
+                          if (defined($ol) and defined($value) and  (length($value) > $ol)) {
+                              $self->db->get_AnalysisDataAdaptor()->store_if_needed($value);
+                          } else {
+                              $value;
+                          }
+                   }
+
+
+             } @$fields ];
+}
+
+sub overflow_limit {
+    return {
+        'key_signature' => 255,
+        'struct_name'   => 255,
```

Comment:
Not sure about overflowing `struct_name`. I see it as a static identifier / parameter name, like `logic_name`, `pipeline_wide_parameters.param_name`, etc. I also think they should be protected, but perhaps differently. Maybe introduce a `size_limit` method in the adaptors that define a maximum allowed size for some fields, and the base `store` method checks the values before doing the INSERT ? It might be possible to get those limits automatically through `DBSQL::BaseAdaptor::_table_info_loader`

Reply:
struct_name needs protection because it's linked to to_analysis_url, which is itself protected. As for param_name, logic_name, and others that get set at init time - I'm not sure it's worth introducing a size limit which would throw an error, since the MySQL error that gets thrown is actually informative in this case.

Reply:
It actually depends how the MySQL server has been configured. mysql-ens-compara-prod-1 complains:
> DBD::mysql::st execute failed: Data too long for column 'logic_name' at row 1 at /home/matthieu/workspace/src/hive/2.5/modules/Bio/EnsEMBL/Hive/DBSQL/StatementHandle.pm line 140.

whereas mysql-ens-reg-prod-4 silently truncates the logic name to 256 characters

Reply:
Remove `struct_name` ?

Reply:
Have been experimenting (not pushed yet) with getting sizes in _table_info_loader(). Straightforward, but for many types COLUMN_SIZE returns null. These are types such as INT or TEXT, where either the size is very large, or that we are less concerned about. The bottom line being that the check against size_limit is not comprehensive without a lot of work (and somehow defining it for every column elsewhere in the code), but it does cover the cases that the ticket and this PR care about